app/templates/_package.json

```diff
@@ -14,7 +14,7 @@
     "node": ">=0.10.0"
   },
   "scripts": {
-    "test": "xo && <% if (nyc) { %>nyc <% } %>ava"
+    "test": "xo && <% if (nyc) { %>nyc <% if (coveralls) { %>--reporter=lcov --reporter=text <% } %><% } %>ava"
```

Comment:
I think it would be better to create a new run script like done here:
https://github.com/sindresorhus/xo/blob/e3eaf881c82957890efa1caab2aee6eb764a3411/package.json#L25

Better separation and then you can simplify the travis file: https://github.com/sindresorhus/xo/blob/e3eaf881c82957890efa1caab2aee6eb764a3411/.travis.yml#L7